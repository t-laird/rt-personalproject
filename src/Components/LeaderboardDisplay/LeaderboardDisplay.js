import React from 'react';
import './LeaderboardDisplay.css';


const LeaderboardDisplay = ({ name, points, position }) => {

  return (
    <div>
      <h3>{position}. {name}: {points}</h3>
    </div>
  ) 
}

export default LeaderboardDisplay;