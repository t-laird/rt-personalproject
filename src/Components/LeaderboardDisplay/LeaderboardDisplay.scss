@import '../../Variables.scss';

.leaderboard-display-component {
  display: flex;
  justify-content: flex-start;
  margin-top: 3px;
  width: 305px;

  .leaderboard-position {
    width: 24px;
    text-align: right;
    margin-right: 10px;
    margin-left: 14px;
    font-weight: 900;
  }

  .leaderboard-name {
    width: 200px;
    color: $blue-dark;
  }

  .leaderboard-points {
    font-weight: 900;
    width: 40px;
    text-align: right;
  }
}
