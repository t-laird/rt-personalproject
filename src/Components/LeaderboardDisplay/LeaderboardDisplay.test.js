import React from 'react';
import { shallow } from 'enzyme';
import LeaderboardDisplay from './LeaderboardDisplay';

describe('leaderboard display tests', () => {
  it('should match the snapshot', () => {
    const renderedApp = shallow(<LeaderboardDisplay />);

    expect(renderedApp).toMatchSnapshot();
  });
});