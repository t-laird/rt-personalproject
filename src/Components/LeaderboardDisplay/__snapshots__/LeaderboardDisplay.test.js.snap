// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`leaderboard display tests should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LeaderboardDisplay />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="leaderboard-position"
>
          .
</h3>,
        <h3
          className="leaderboard-name"
/>,
        <h3
          className="leaderboard-points"
/>,
      ],
      "className": "leaderboard-display-component",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            ".",
          ],
          "className": "leaderboard-position",
        },
        "ref": null,
        "rendered": Array [
          undefined,
          ".",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "leaderboard-name",
        },
        "ref": null,
        "rendered": null,
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "leaderboard-points",
        },
        "ref": null,
        "rendered": null,
        "type": "h3",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="leaderboard-position"
>
            .
</h3>,
          <h3
            className="leaderboard-name"
/>,
          <h3
            className="leaderboard-points"
/>,
        ],
        "className": "leaderboard-display-component",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              ".",
            ],
            "className": "leaderboard-position",
          },
          "ref": null,
          "rendered": Array [
            undefined,
            ".",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "leaderboard-name",
          },
          "ref": null,
          "rendered": null,
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "leaderboard-points",
          },
          "ref": null,
          "rendered": null,
          "type": "h3",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
